### ISSUE TITLE
Collabora Office can't open or save via FileProvider ( *.ODT, *. ODS, *.ODP)

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/ios-app#557]
[Collabora Office](https://www.collaboraoffice.com/release-news/collabora_office_first_free_open_source_office-app_for_ios/) can't open or save via FileProvider ( *.ODT, *. ODS, *.ODP)

They use the FileProvider, but *.ODT files are shown greyed out:

![2019-11-06_19_17_56](https://user-images.githubusercontent.com/214010/68330105-8e5bdc00-00ca-11ea-9d6d-9f38bec8895d.png)

This seems only a problem in the ownCloud implementation, I don't have this problem with other FileProviders.

### Client
**iOS version:**
13.2 (17B84)

**ownCloud app version:**
1.1.2 build 144 (app: de68a46, sdk: edc8c00)

**Device model:**
iPhone 11 Pro Max

**Comment by felix-schwarz** [comment]
Collabora Office uses `UIDocumentBrowserViewController` for this interface, which is [initialized with an array of compatible UTIs](https://developer.apple.com/documentation/uikit/uidocumentbrowserviewcontroller/2874203-initforopeningfileswithcontentty?language=objc) that express which file types can be picked.

In the FileProvider, the file's MIME-Type retrieved from the ownCloud server is converted to an UTI using an API provided by iOS. So something must go wrong there:

- ✅ according to [Wikipedia](https://en.wikipedia.org/wiki/OpenDocument), the MIME-Type for ODT files is `application/vnd.oasis.opendocument.text` while the UTI is `org.oasis.opendocument.text`.
- ✅ the ownCloud server returns `application/vnd.oasis.opendocument.text` as MIME-Type for `Example.odt`
- ⚠️ the iOS API *in the Simulator* converts that to the UTI `org.oasis-open.opendocument.text`, which is a little different from what Wikipedia mentions
- 🚨 the iOS API *on a device* converts that to the UTI `dyn.agq80c6durvy0g2pyrf106p50r3wc655bsry1gpxtsbw063dtqr4043pssu1hk3p2su`, which I'm pretty certain will never match an UTI listed by Collabora Office as compatible.

Why iOS behaves so different when it comes to UTIs in the Simulator and on a device I don't know.

So what I did is I tuned the code converting MIME-Types to UTIs to return …
- ⚠️ `application/vnd.oasis.opendocument.text` which should be formally correct but will still result in the file not being selectable in Collabora Office
- ✅ `org.oasis-open.opendocument.text` which is what the iOS Simulator returned and also what Collabora Office seems to specify. With that UTI, I can now open the file in Collabora Office.

I'll put together a small PR to fix the problem.
